typing/ctype.ml

```diff
@@ -2455,14 +2718,14 @@ and mcomp_type_decl type_pairs env p1 p2 tl1 tl2 =
       | Type_record (lst,r), Type_record (lst',r') when r = r' ->
           mcomp_list type_pairs env tl1 tl2;
           mcomp_record_description type_pairs env lst lst'
-      | Type_variant v1, Type_variant v2 ->
+      | Type_variant (v1,r), Type_variant (v2,r') when r = r' ->
```

Comment:
This is really part of [upstream 10252](https://github.com/ocaml/ocaml/pull/10252), which added `variant_representation`, and which we are building on.  (`ocaml-jst` has since rebased past 10252, so this has fallen out of our diff since you looked at it).